### PR TITLE
NIAD-1003 Mock MHS to require wait-for-response header

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/controller/MhsMockController.java [hunk 1]
```diff
@@ -41,7 +41,8 @@
         try {
             String interactionId = Optional.ofNullable(headers.get("interaction-id")).orElse("");
             String correlationId = Optional.ofNullable(headers.get("correlation-id")).orElse("");
-            return mockMhsService.handleRequest(interactionId, correlationId, mockMhsMessage);
+            String waitForResponse = Optional.ofNullable(headers.get("wait-for-response")).orElse("");
```
Comment (mmackiewicz1) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/controller/MhsMockController.java:44]: Could we use a constant "StringUtils.EMPTY" instead?

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java [hunk 2]
```diff
@@ -41,11 +42,16 @@
     private final InputStream inputStream3 = this.getClass().getClassLoader().getResourceAsStream("InternalServerError.html");
     private final String internalServerErrorResponse = IOUtils.toString(inputStream3, StandardCharsets.UTF_8);
 
-    public ResponseEntity<String> handleRequest(String interactionId, String correlationId, String mockMhsMessage) {
+    public ResponseEntity<String> handleRequest(String interactionId, String correlationId, String waitForResponse, String mockMhsMessage) {
         headers.setContentType(MediaType.TEXT_HTML);
         objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
         objectMapper.configure(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES, true);
 
+        if(!waitForResponse.equals("false")) {
```
Comment (mmackiewicz1) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:50]: Missing empty space after "if". Also "false" could be replaced with Boolean constant.

**Review by Broom94 — DISMISSED** [review]

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/controller/MhsMockController.java (hunk 1 above)
Comment (mattd-kainos) [reply mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/controller/MhsMockController.java]: fixed

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java (hunk 2 above)
Comment (mattd-kainos) [reply mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:50]: applied formatting, will leave string literal. Only used once and it is more clear that this value exactly is required.

**Review by Broom94 — APPROVED** [review]

**Review by ElliotB-96 — APPROVED** [review]

**Pull request merged**